### PR TITLE
Combining one-shot with snapshot creation.

### PR DESCRIPTION
**Pull request by rachely3n**
As discussed, we should only allow snapshot creation when one-shot is enabled.
The changes involved have 2 portions.

UI
* One-shot has own section with two checkboxes: Enabled and Create snapshot?

Backend
* Because the OnceRetentionStrategy terminates the One-shot instance via _terminate in ComputeEngineInstance, had to remove the snapshot logic from ComputeEngineComputer to ComputeEngineInstance.
* Above point breaks integration tests. Fixed them by polling for node removal, which will guarantee snapshot was created (given that there are not errors).

### CODE REVIEW

**Comment by craigatgoogle** [comment]
If we're going to require one-shot being enabled as a requirement for snapshotting, then the descriptor for InstanceConfiguration should have validation logic added to it which asserts this.

**Comment by rachely3n** [comment]
Ah, good catch. Working on it.

**Comment by rachely3n** [comment]
Updated, both front-end and back-end validation added.

File: src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java [hunk 1]
```diff
@@ -854,5 +860,14 @@ public FormValidation doCheckWindowsPrivateKeyCredentialsId(@AncestorInPath Jenk
             }
             return FormValidation.ok();
         }
+
+        public FormValidation doCheckCreateSnapshot(@AncestorInPath Jenkins context,
+                                                                             @QueryParameter boolean value,
+                                                                             @QueryParameter("oneShot") boolean oneShot) {
+            if (!oneShot && value) {
+                return FormValidation.error("One-shot must be enabled to create snapshots");
```
Comment (craigatgoogle) [inline src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java]: This error message needs to live in Message.properties for i18ln.

File: pom.xml [hunk 2]
```diff
@@ -11,7 +11,7 @@
     <version>1.0.11-SNAPSHOT</version>
     <packaging>hpi</packaging>
     <properties>
-        <jenkins.version>2.7.3</jenkins.version>
+        <jenkins.version>2.107.1</jenkins.version>
```
Comment (craigatgoogle) [inline pom.xml]: This isn't really related to the changes in this PR. It should be pulled out into its own PR.

Comment (rachely3n) [reply pom.xml]: Removed

File: src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java (hunk 1 above)
Comment (rachely3n) [reply src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java]: Moved
